cmplxs/cmplxs.go

```diff
@@ -0,0 +1,954 @@
+			if f(val) {
+				inds = append(inds, i)
+			}
+		}
+		return inds, nil
+	}
+
+	// Otherwise, find the first k elements
+	nFound := 0
+	for i, val := range s {
+		if f(val) {
+			inds = append(inds, i)
+			nFound++
+			if nFound == k {
+				return inds, nil
+			}
+		}
+	}
+	// Finished iterating over the loop, which means k elements were not found
+	return inds, errors.New("cmplxs.Find: insufficient elements found")
+}
+
+// HasNaN returns true if the slice s has any values that are NaN and false
+// otherwise.
+func HasNaN(s []complex128) bool {
+	for _, v := range s {
+		if cmplx.IsNaN(v) {
+			return true
+		}
+	}
+	return false
+}
+
+// Hermitian generates the hermitian (conjugate transpose) of s with r rows and c cols,
+// and stores in s.
+// It panics if the length of s is not equal to r * c.
+func Hermitian(r, c int, s []complex128) {
+	if (r * c) != len(s) {
+		panic("cmplxs.Hermitian: rows & cols do not match with length of s")
+	}
+	dst := make([]complex128, len(s))
+	for i := 0; i < r; i++ {
+		for j := 0; j < c; j++ {
+			dst[i+r*j] = cmplx.Conj(s[i*c+j])
+		}
+	}
+	copy(s, dst)
+}
+
+// HermitianTo generates the hermitian (conjugate transpose) of s with r rows and c cols,
+// and stores in dst.
+// It panics if the length of s is not equal to r * c and if the length of s and dst are
+// not equal.
+func HermitianTo(r, c int, dst, s []complex128) []complex128 {
+	if (r * c) != len(s) {
+		panic("cmplxs.HermitianTo: rows & cols do not match with length of s")
+	}
+	if len(s) != len(dst) {
+		panic("cmplxs.HermitianTo: length of s and dst do not match")
+	}
+	for i := 0; i < r; i++ {
+		for j := 0; j < c; j++ {
+			dst[i+r*j] = cmplx.Conj(s[i*c+j])
+		}
+	}
+	return dst
+}
+
+// Imag calculates imag(s) and stores in dst. dst is returned.
+// It panics if the lengths of dst and s are not equal.
+func Imag(dst []float64, s []complex128) []float64 {
+	if len(s) != len(dst) {
+		panic("cmplxs.Imag: length of s and dst do not match")
+	}
+	if len(s) == 0 {
+		return dst
+	}
+	for i, v := range s {
+		dst[i] = imag(v)
+	}
+	return dst
+}
+
+// L1Dist returns the 1-Norm of t - s
+func L1Dist(s, t []complex128) float64 {
+	var norm float64
+	switch {
+	case len(s) > len(t):
+		for i, v := range t {
+			norm += cmplx.Abs(v - s[i])
+		}
+	default:
+		for i, v := range s {
+			norm += cmplx.Abs(t[i] - v)
+		}
+	}
+	return norm
+}
+
+// LinfDist returns the inf-Norm of t - s
+func LinfDist(s, t []complex128) float64 {
+	var norm float64
+	if len(s) == 0 {
+		return 0
+	}
+	norm = cmplx.Abs(t[0] - s[0])
+	switch {
+	case len(s) > len(t):
+		for i, v := range t[1:] {
+			absDiff := cmplx.Abs(v - s[i+1])
+			if absDiff > norm || math.IsNaN(norm) {
+				norm = absDiff
+			}
+		}
+	default:
+		for i, v := range s[1:] {
+			absDiff := cmplx.Abs(t[i+1] - v)
+			if absDiff > norm || math.IsNaN(norm) {
+				norm = absDiff
+			}
+		}
+	}
+	return norm
+}
+
+// L1Norm returns the 1-Norm of x
+func L1Norm(x []complex128) (sum float64) {
```

Comment:
Why are there a bunch of Norm functions. In floats, we have one with a norm selector parameter. Please reflect that here.